central/image/service/service_impl.go

```diff
@@ -414,31 +421,43 @@ func (s *serviceImpl) EnrichLocalImageInternal(ctx context.Context, request *v1.
 		if exists && !forceScanUpdate && !forceSigVerificationUpdate {
 			return internalScanRespFromImage(existingImg), nil
 		}
+
+		if exists {
+			imgExists = true
+		}
 	}
 
 	img := &storage.Image{
-		Id:             imgID,
-		Name:           request.GetImageName(),
+		Id:   imgID,
+		Name: request.GetImageName(),
+		// 'Names' must be populated to enable cache hits in central AND sensor
+		Names:          []*storage.ImageName{request.GetImageName()},
```

Comment:
I'm not as familiar with this field. Does this deprecate `Name` at all?

Reply:
The `Names` field does have a comment indicating that long-term it may replace `Name` (in `proto/storage/image.proto`). I don't know the history or if we're tracking that change elsewhere.

Without setting this the cache is essentially skipped (we store images in the cache but never pull them out).